utils/src/bitmap/mod.rs

```diff
@@ -0,0 +1,1988 @@
+
+    /// A chunk of all 0s.
+    const EMPTY_CHUNK: [u8; N] = [0u8; N];
+
+    /// A chunk of all 1s.
+    const FULL_CHUNK: [u8; N] = [u8::MAX; N];
+
+    /* Constructors */
+
+    /// Create a new empty bitmap.
+    pub fn new() -> Self {
+        #[allow(path_statements)]
+        Self::_CHUNK_SIZE_NON_ZERO_ASSERT; // Prevent compilation for N == 0
+
+        let bitmap = VecDeque::from([Self::EMPTY_CHUNK]);
+        Self {
+            chunks: bitmap,
+            next_bit: 0,
+        }
+    }
+
+    // Create a new empty bitmap with the capacity to hold `size` bits without reallocating.
+    pub fn with_capacity(size: u64) -> Self {
+        #[allow(path_statements)]
+        Self::_CHUNK_SIZE_NON_ZERO_ASSERT; // Prevent compilation for N == 0
+
+        let num_chunks = Self::num_chunks_at_size(size).get();
+        let mut chunks = VecDeque::with_capacity(num_chunks);
+        // Invariant: chunks is never empty
+        chunks.push_back(Self::EMPTY_CHUNK);
+        Self {
+            chunks,
+            next_bit: 0,
+        }
+    }
+
+    /// Create a new bitmap with `size` bits, with all bits set to 0.
+    pub fn zeroes(size: u64) -> Self {
+        #[allow(path_statements)]
+        Self::_CHUNK_SIZE_NON_ZERO_ASSERT; // Prevent compilation for N == 0
+
+        let num_chunks = Self::num_chunks_at_size(size).get();
+        let mut chunks = VecDeque::with_capacity(num_chunks);
+        for _ in 0..num_chunks {
+            chunks.push_back(Self::EMPTY_CHUNK);
+        }
+        Self {
+            chunks,
+            next_bit: size % Self::CHUNK_SIZE_BITS,
+        }
+    }
+
+    /// Create a new bitmap with `size` bits, with all bits set to 1.
+    pub fn ones(size: u64) -> Self {
+        #[allow(path_statements)]
+        Self::_CHUNK_SIZE_NON_ZERO_ASSERT; // Prevent compilation for N == 0
+
+        let num_chunks = Self::num_chunks_at_size(size).get();
+        let mut chunks = VecDeque::with_capacity(num_chunks);
+        for _ in 0..num_chunks {
+            chunks.push_back(Self::FULL_CHUNK);
+        }
+        let mut result = Self {
+            chunks,
+            next_bit: size % Self::CHUNK_SIZE_BITS,
+        };
+        // Clear trailing bits to maintain invariant
+        result.clear_trailing_bits();
+        result
+    }
+
+    /* Length */
+
+    /// Return the number of bits currently stored in the bitmap.
+    #[inline]
+    pub fn len(&self) -> u64 {
+        ((self.chunks.len() as u64 - 1) * Self::CHUNK_SIZE_BITS) + self.next_bit
+    }
+
+    /// Returns true if the bitmap is empty.
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    // Get the number of chunks currently in the bitmap.
+    fn chunks_len(&self) -> usize {
+        self.chunks.len()
+    }
+
+    /* Getters */
+
+    /// Get the value of a bit.
+    ///
+    /// # Warning
+    ///
+    /// Panics if the bit doesn't exist.
+    #[inline]
+    pub fn get(&self, bit_offset: u64) -> bool {
```

Comment:
nit: I think we could change all these `bit_offset` args to `offset` (I think this is better than changing the function that accepts `index` to `chunk_index` but could be convinced to go that way).

Reply:
I changed `bit_offset` to `bit` and `chunk_index` to `chunk`. Also removed a bunch of other `_offset` that aren't really needed.